src/fsharp/TypeChecker.fs

```diff
@@ -6386,32 +6388,61 @@ and TcIndexerThen cenv env overallTy mWholeExpr mDot tpenv wholeExpr e1 indexArg
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.One _;SynIndexerArg.Two _], _, _)    -> Some (sliceOpPath, "GetArraySlice3DFixedSingle2", expandedIndexArgs)
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.Two _;SynIndexerArg.One _], _, _)    -> Some (sliceOpPath, "GetArraySlice3DFixedSingle3", expandedIndexArgs)
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.One _;SynIndexerArg.One _;SynIndexerArg.Two _], _, _)    -> Some (sliceOpPath, "GetArraySlice3DFixedDouble1", expandedIndexArgs)
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.One _;SynIndexerArg.Two _;SynIndexerArg.One _], _, _)    -> Some (sliceOpPath, "GetArraySlice3DFixedDouble2", expandedIndexArgs)
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.One _;SynIndexerArg.One _], _, _)    -> Some (sliceOpPath, "GetArraySlice3DFixedDouble3", expandedIndexArgs)
-                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.Two _;SynIndexerArg.Two _;SynIndexerArg.Two _], _, _)      -> Some (sliceOpPath, "GetArraySlice4D", expandedIndexArgs)
+                | true, false, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _], _, _)                                                                    -> Some (sliceOpPath, "GetStringSlice", expandedIndexArgs)
+                | true, false, SynExpr.DotIndexedGet (_, [SynIndexerArg.One _], _, _)                                                                    -> Some (indexOpPath, "GetString", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _], _, _)                                                                    -> Some (sliceOpPath, "GetArraySlice", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.One _;SynIndexerArg.Two _], _, _)                                                -> Some (sliceOpPath, "GetArraySlice2DFixed1", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.One _], _, _)                                                -> Some (sliceOpPath, "GetArraySlice2DFixed2", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.Two _], _, _)                                                -> Some (sliceOpPath, "GetArraySlice2D", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.Two _;SynIndexerArg.Two _;SynIndexerArg.Two _], _, _)                            -> Some (sliceOpPath, "GetArraySlice3D", expandedIndexArgs)
+                | false, true, SynExpr.DotIndexedGet (_, [SynIndexerArg.One _;SynIndexerArg.Two _;SynIndexerArg.Two _], _, _)                            when fixedIndex3d4dEnabled -> Some (sliceOpPath, "GetArraySlice3DFixedSingle1", expandedIndexArgs)
```

Comment:
I think we should make a single branch on a single condition with `fixedIndex3d4dEnabled` and then check for the expected 3d4d arrays.
i.e.
```fsharp
match isString, isArray, wholeExpr with
...
| _ when fixedIndex3d4dEnabled ->
    (* 3d4d array handling for fixed indices *)
    ...
```